app/workers/rent_worker.rb

```diff
@@ -0,0 +1,11 @@
+require 'sidekiq-scheduler'
+
+class RentWorker
+  include Sidekiq::Worker
+
+  def perform
+    Rent.all.each do |rent|
```

Comment:
There are some issues here that I want to talk about.

- Firstly, I want to talk about the `.all` method. This method loads the whole table in memory. This could bring you some problems with the memory, as you are instantiating every row of the table as a new ActiveRecord object. There's a solution for that, which is using the method find_each as the doc says. Check it out. http://guides.rubyonrails.org/active_record_querying.html#retrieving-multiple-objects-in-batches. This methods makes queries in batches, so that it does not instantiate every record at the same time

- Secondly, I want to talk about encapsulation and responsibilities. ActiveRecord by default generates getters for every attribute. This is really nice, as you don't have to declare those, but sometimes is not really good. Sometimes you end up having logic that it's the model's responsibility, just because you can access to its attributes. In this case, you are accessing the attribute to and comparing it with the current time. This has some meaning for the rent which is to check if the rent is expired. So probably you will need to know in some other code when the rent is expired, and you will make some kind of copy and paste of that method. As requirements change, you may need to change the way a rent says that it is expired, so we will have to find out every place that you copy and pasted that logic.

- Regarding performance, there's another thing that I would like to talk about. Here, you are loading unnecessary rents, that then you are filtering in memory, leading to ignore, maybe, lots of records. Probably a better approach would be to load only the expired rents from the database, using where clauses.

- Lastly, I want to talk about some issues about requirements or functionality. What happens if you run this worker multiple times? It will always send an email to the rents that are expired and I don't think that's a great idea. I don't like the idea of having a guy sending me an email every day after my rent has expired. Even more if I have already returned my book. So you need to have some boolean or condition that let's you check if you have to send the email, even though the rent is expired.

Reply:
I don't expect that you fix all this things, but I wanted to explain you this things, because I consider them really important.

For this pr, it will be nice if you fix the first to items